cmd/integrationArtifactUpload_test.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type integrationArtifactUploadMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newIntegrationArtifactUploadTestsUtils() integrationArtifactUploadMockUtils {
+	utils := integrationArtifactUploadMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunIntegrationArtifactUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Create Test", func(t *testing.T) {
+
+		tempDir, tmpErr := ioutil.TempDir("", "")
```

Comment:
Instead of working with a tempory directory the file mocking facilities can be used.  File mock is already prepared above.